unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarModifiers/AvatarModifierArea.cs

```diff
@@ -152,8 +170,84 @@ private void ApplyCurrentModel()
                 OnAvatarEnter += modifier.ApplyModifier;
                 OnAvatarExit += modifier.RemoveModifier;
             }
+
+            // Set excluded colliders
+            excludedColliders = GetExcludedColliders(cachedModel);
+
+            if (cachedModel.excludeIds != null && cachedModel.excludeIds.Length > 0)
+            {
+                DataStore.i.player.ownPlayer.OnChange += OwnPlayerOnOnChange;
+                DataStore.i.player.otherPlayers.OnAdded += OtherPlayersOnOnAdded;
+            }
+        }
+    }
+
+    public override int GetClassId()
+    {
+        return (int)CLASS_ID_COMPONENT.AVATAR_MODIFIER_AREA;
+    }
+
+    private static HashSet<Collider> GetExcludedColliders(in Model componentModel)
+    {
+        string[] excludeIds = componentModel?.excludeIds;
+        if (excludeIds == null || excludeIds.Length == 0)
+        {
+            return null;
+        }
+
+        var ownPlayer = DataStore.i.player.ownPlayer.Get();
+        var otherPlayers = DataStore.i.player.otherPlayers;
+
+        HashSet<Collider> result = new HashSet<Collider>();
+        for (int i = 0; i < excludeIds.Length; i++)
+        {
+            if (ownPlayer != null && excludeIds[i] == ownPlayer.id)
+            {
+                result.Add(ownPlayer.collider);
+            }
+            else if (otherPlayers.TryGetValue(excludeIds[i], out Player player))
+            {
+                result.Add(player.collider);
+            }
         }
+        return result;
+    }
+
+    private void OtherPlayersOnOnAdded(string id, Player player)
+    {
+        RefreshExclusionList(player);
+    }
+
+    private void OwnPlayerOnOnChange(Player current, Player previous)
+    {
+        RefreshExclusionList(current);
+    }
+
+    private void OnOtherPlayersRemoved(string id, Player player)
+    {
+        excludedColliders?.Remove(player.collider);
     }
 
-    public override int GetClassId() { return (int) CLASS_ID_COMPONENT.AVATAR_MODIFIER_AREA; }
+    private void RefreshExclusionList(Player player)
+    {
+        string[] excludeIds = cachedModel?.excludeIds;
+        if (excludeIds == null || excludeIds.Length == 0)
+        {
+            return;
+        }
+
+        if (!excludeIds.Contains(player.id))
+        {
+            return;
+        }
+
+        if (excludedColliders == null)
+        {
+            excludedColliders = new HashSet<Collider>() { player.collider };
+        }
+        else if (!excludedColliders.Contains(player.collider))
```

Comment:
I think this check can be avoided right? since the collider can only exist once in the hashset

Reply:
yeah, you are right! I thought that `add` could throw an exception if element is already there (as Dictionary does) but it doesn't.
Changed here: https://github.com/decentraland/unity-renderer/pull/1853/commits/343a0512997ef9de3670de0da08f71f8c6a0780b